lib/puppet/type/file/content.rb

```diff
@@ -220,6 +221,19 @@ def get_from_source(source_or_content, &block)
       end
     end
 
+    def get_from_http_source(source, &block)
+      Puppet::Util::HttpProxy.request_with_redirects(URI(source), :get, &block)
```

Comment:
I think I see what you mean. Turns out that is no agent option after all. In fact, I disbelieve that there is *any* code currently in place to override this for any HTTP request.

I believe that this is fine. The above line can be patched later to pass either an agent option or a resource parameter value to the method to pick a custom limit.